azkaban-common/src/test/java/azkaban/user/XmlUserManagerTest.java

```diff
@@ -73,6 +78,59 @@ public void testDoNotExist() throws Exception {
     fail("XmlUserManager should throw an exception when the file doesn't exist");
   }
 
+  /**
+   * Test auto reload of user XML
+   */
+  @Test
+  public void testAutoReload() throws Exception {
+    final Props props = new Props(this.baseProps);
+    final URL configURL = Resources.getResource("test-conf/azkaban-users-test1.xml");
+    final String path = configURL.getPath();
+    props.put(XmlUserManager.XML_FILE_PARAM, path);
+
+    UserManager manager;
+    try {
+      manager = new XmlUserManager(props);
+    } catch (final RuntimeException e) {
+      fail("Should have found the xml file");
+      return;
+    }
+
+    // Get the user8 from existing XML with password == password8
+    User user8 = manager.getUser("user8", "password8");
+
+    // Modify the password for user8
+    // TODO : djaiswal : Find a better way to modify XML
+    List<String> lines = new ArrayList<>();
+    List<String> origLines = new ArrayList<>();
+    for (String line : Files.readAllLines(Paths.get(path))) {
+      origLines.add(line);
+      if (line.contains("password8")) {
+        lines.add(line.replace("password8", "passwordModified"));
+      } else {
+        lines.add(line);
+      }
+    }
+    // Update the file
+    Files.write(Paths.get(path), lines);
+
+    // This should reload the XML file. Check for user8 again
+    try {
+      user8 = manager.getUser("user8", "password8");
+    } catch (final UserManagerException e) {
+      System.out.println("USer8 has updated password" + e.getMessage());
+    }
+
+    try {
+      user8 = manager.getUser("user8", "passwordModified");
```

Comment:
How does the test make sure the watcher thread has processed the event before this point?

Reply:
This is a good point. I will think of a way to do it.